### PR TITLE
Use pnpx for Wrangler

### PR DESCRIPTION
**Pull request by penalosa**
## What this PR solves / how to test

Fixes #000

TLDR asset-worker + router-worker used Wrangler to deploy, but workers-shared had no dependency on Wrangler. I couldn't add a dependency because that would be circular (since Wrangler depends on workers shared), so I used pnpx to run the latest Wrangler version

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Tests included
  - [x] Tests not necessary because: change to deploy script
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [ ] I don't know
  - [ ] Required
  - [x] Not required because: change to deploy script, not covered by e2e
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Changeset included
  - [ ] Changeset not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [x] Documentation not necessary because: bugfix to deploy script

### CODE REVIEW

**Review by GregBrimble — APPROVED** [review]

**Comment by petebacondarwin** [comment]
Why is this needed?

**Review by lrapoport-cf — APPROVED** [review]
